src/main/java/core/specialevents/PowerfulEventPredictionAnalyzer.java

```diff
@@ -0,0 +1,141 @@
+        int id = position.getSpielerId();
+        Player p = analyse.getPlayer(id);
+        if (p.hasSpeciality(Speciality.POWERFUL)) {
+
+            switch (position.getId()) {
+                case IMatchRoleID.leftForward:
+                case IMatchRoleID.centralForward:
+                case IMatchRoleID.rightForward:
+
+                    if (position.getTaktik() == IMatchRoleID.NORMAL) {
+                        getPowerfulNormalForward(position);
+                    }
+                    break;
+
+                case IMatchRoleID.rightInnerMidfield:
+                case IMatchRoleID.centralInnerMidfield:
+                case IMatchRoleID.leftInnerMidfield:
+                    if (position.getTaktik() == IMatchRoleID.DEFENSIVE) {
+                        getSittingMidfielder(position);
+                    }
+                    break;
+            }
+        }
+    }
+
+    // PDIMs reduces goal chances of opponent teams
+    // => chance probability < 0 to display it in opponents column
+    // => goal probability > 0  to reduce opponents goals
+    private void getSittingMidfielder( MatchRoleID position, MatchRoleID opponentScorer) {
+        Player p = analyse.getPlayer(position.getSpielerId());
+        Player op = analyse.getOpponentPlayer(opponentScorer.getSpielerId());
+        SpecialEventsPrediction se = SpecialEventsPrediction.createIfInRange(position, SpecialEventType.PDIM,
+                -.1, 10, -10,
+                p.getDEFskill() + p.getKondition() - op.getSCskill() - op.getKondition()
+        );
+        if (se != null) {
+            double goalP = analyse.getOpponentGoalProbability(opponentScorer);
+            if (goalP > 0) {
+                se.setInvolvedOpponentPosition(opponentScorer);
+                se.setGoalProbability(-se.getChanceCreationProbability() * goalP);
+                analyse.addSpecialEventPrediction(se);
+            }
+        }
+    }
+
+    private void getSittingMidfielder( MatchRoleID position) {
+        double overcrowdingFactor = getOvercrowding(position, IMatchRoleID.rightInnerMidfield, IMatchRoleID.leftInnerMidfield, IMatchRoleID.DEFENSIVE);
+
+        // Any opponent player, except keeper, could be involved
+        for (int i = IMatchRoleID.rightBack; i <= IMatchRoleID.leftForward; i++) {
+            MatchRoleID opponentScorer = analyse.getOpponentPosition(i);
+            if (opponentScorer.getSpielerId() != 0) {
+                getSittingMidfielder( position, opponentScorer);
+            }
+        }
+    }
+
+    private double getOvercrowding(MatchRoleID position, int right, int left, byte taktik) {
+        int overcrowding = 0;
+        for (int i = right; i <= left; i++) {
+            if (i != position.getId()) {
+                MatchRoleID mid = this.analyse.getPosition(i);
+                if (mid.getTaktik() == taktik) {
+                    Player p = analyse.getPlayer(mid.getSpielerId());
+                    if (p.hasSpeciality(Speciality.POWERFUL)) {
```

Comment:
a lot of nested `if`, maybe you can split this methods in order to reduce complexity